### PR TITLE
RN-52: RN: Make create post and upload file actions optimistic

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
This PR allows for posts to be submitted optimistically.

#### Ticket Link
https://mattermost.atlassian.net/browse/RN-52

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to the drivers

#### Test Information
This PR was tested on:
iPhone 6 iOS 10
iPhone 7 iOS 10
Galaxy s7 - Android 6.0 API 23
@jarredwitt

### CODE REVIEW

File: src/client/client4.js [hunk 1]
```diff
@@ -729,12 +728,17 @@ export default class Client4 {
...
+            contentType += `; boundary=${formBoundary}`;
+        }
+
         return this.doFetch(
             `${this.getFilesRoute()}`,
             {
                 method: 'post',
                 headers: {
-                    'Content-Type': `multipart/form-data; boundary=${formBoundary}`
+                    'Content-Type': contentType
```
Comment (enahum) [inline src/client/client4.js]: @jarredwitt why this change?

Comment (jarredwitt) [inline src/client/client4.js]: Android fetch will not work if formBoundary is a part of the content-type.
